packages/block-library/src/image/image.js

```diff
@@ -498,15 +500,12 @@ export default function Image( {
 			naturalHeight < naturalWidth ? MIN_SIZE : MIN_SIZE / ratio;
 
 		// With the current implementation of ResizableBox, an image needs an
-		// explicit pixel value for the max-width. In absence of being able to
-		// set the content-width, this max-width is currently dictated by the
-		// vanilla editor style. The following variable adds a buffer to this
-		// vanilla style, so 3rd party themes have some wiggleroom. This does,
-		// in most cases, allow you to scale the image beyond the width of the
-		// main column, though not infinitely.
-		// @todo It would be good to revisit this once a content-width variable
-		// becomes available.
-		const maxWidthBuffer = maxWidth * 2.5;
+		// explicit pixel value for the max-width.
+		// The block will try to use a wide layout size defined in the `theme.json`
+		// or fall back to the editor max-width + some wiggle-room.
+		const maxWidthBuffer = wideSize
+			? parseInt( wideSize, 10 )
```

Comment:
`wideSize` might not be in `px` and that could result to quite unexpected results. Also shouldn't we take into account if an image is inside a container with `wideSize` set?

Reply:
> wideSize might not be in px and that could result to quite unexpected results.

I didn't know all the theme.json files I checked had `px` specified. But it shouldn't be problem all the following will return `1000`:

```js
parseInt('1000rem', 10);
parseInt('1000', 10);
parseInt(1000, 10);
```

> Also shouldn't we take into account if an image is inside a container with wideSize set?

The current implementation has the same issue. Ideally, we want ResizableBox to have `max-width: 100%`, but currently, it only accepts px values. Here're some details: https://github.com/WordPress/gutenberg/pull/11846#issuecomment-439033037

Reply:
>But it shouldn't be problem all the following will return 1000:

Yes, but if we have something like `50%`, it will make the max width `50px`. Maybe we should try to improve when we only have a px value? In addition if the value here is smaller than than the viewport it keeps the image contained and I noticed some distortion of the height in that case when I stop dragging.

https://user-images.githubusercontent.com/16275880/159717776-d43521a6-94f3-4278-945d-f26736f4bf3c.mov

Reply:
Good point. I'll look into this tomorrow. Thanks, Nik!